refl1d/probe/probe.py

```diff
@@ -856,7 +856,7 @@ def _set_calc(self, T, L):
         idx = np.argsort(Q)
         self.calc_T = T[idx]
         self.calc_L = L[idx]
-        self.calc_Qo = Q[idx]
+        self.calc_Qo = np.unique(Q[idx])
```

Comment:
This breaks the link between the length of `calc_Qo` and the lengths of `calc_T` and `calc_L`. That it is working at all means that `calc_T` and `calc_L` are not used anywhere and can be removed.

Reply:
Okay, not strictly true. It is used for theta offset when generating a new calc_Q.

Perhaps we should drop the unique, and simply allow the occasional duplicate.

Theta offset does not protect against duplicate Q values appearing, so the resolution code has to be robust against duplicates.

Separate issue: I believe resolution expects non-decreasing Q. We should therefore be sorting the Q values returned from the Q property. Perhaps we could cache the unique Q for the current theta_offset setting.